dev/build_docker_images.sh

```diff
@@ -4,35 +4,49 @@
 #   os
 #   docker_username
 #   docker_password
+#   GITHUB_REF
 
 python_version="3.8"
-tags=""
 version="$(cat version)"
 
+set -ue
+
+git_tag="${GITHUB_REF#refs/tags/v*}"
+if [[ "${git_tag}" != "${version}" ]]; then
+  >&2 echo "ERROR: Mismatch between Git tag ${git_tag} and version file ${version}" 
+  exit 1
+fi
+
 effective_version="${os}_version"
 suffix="${!effective_version}"
-latest="egnyte/gitlabform:latest-${suffix}"
+image_name="egnyte/gitlabform:${version}-${suffix}"
+os_latest="egnyte/gitlabform:latest-${suffix}"
 
-docker pull "${latest}" || echo "no cache is available"
+docker pull "${os_latest}" || echo "no cache is available"
 docker build \
   --build-arg PY_VERSION="${python_version}" \
   --build-arg OS_VERSION="${!effective_version}" \
   --file "dev/${os}.Dockerfile" \
-  --tag "${latest}" .
-tags="$tags ${latest}"
+  --tag "${image_name}" .
+
+tags=( "${image_name}" )
 
-docker tag "${latest}" "egnyte/gitlabform:${version}-${suffix}"
-tags="$tags egnyte/gitlabform:${version}-${suffix}"
+# TODO: something more clever here maybe?
+# TODO: do we need to validate the version?
```

Comment:
I was thinking about keeping the GitHub Actions job "Skip if this is not a final release" but modify it to pass info if this a final release or not to the bash script. This way we will use the existing Python logic for deciding if this is a final release or not and not have to reimplement it in bash.

Reply:
That would be perfect. :)

Reply:
Scipt is there. I find the logic of [packaging module](https://packaging.pypa.io/en/latest/version.html
) (or maybe PEP-440) quite baffling though :-)

Namely, I was in no way able to make it raise an exception:

```
$ ./dev/check_release_type.py -V 1.2.3
public
$ ./dev/check_release_type.py -V 1.2.3rc1
pre
$ ./dev/check_release_type.py -V 1.2.3dev
dev
$ ./dev/check_release_type.py -V 1.2.3post1
post
$ ./dev/check_release_type.py -V 1.2.3post1-123.123.123ABC-..
public
$ ./dev/check_release_type.py -V thisisratherweirdisntit
public
$ ./dev/check_release_type.py -V thisisratherweirdisntit?
public
$

```

Reply:
True, it is strange.. If I am reading https://www.python.org/dev/peps/pep-0440/#version-scheme correctly then a valid PEP 440 version has to contain at least one digit... And `packaging` declares that `parse` can throw an `packaging.version.InvalidVersion` exception...

But anyway, I would rather check only if it `is_prerelease` or anything else. As in the latter category are both public as well as post release and we DO sometimes release posts and they should be treated as the latest.

Reply:
Changed per suggestion.